plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Transformer } from './index';
+
+// requires repo
+export const addGitFeedbackLink = (): Transformer => {
+  return dom => {
+    const contentContainer = dom.querySelector(
+      '.md-content__inner',
+    ) as HTMLElement;
+    const sourceAnchor = dom.querySelector(
+      '.md-content__inner>a',
+    ) as HTMLAnchorElement;
+    // don't show if Gitlab or Github hosting not specified
+    if (!sourceAnchor || !sourceAnchor.href) {
+      return dom;
+    }
+    let gitHost = '';
+
+    if (sourceAnchor.href.includes('gitlab')) {
+      gitHost = 'gitlab';
+    } else if (sourceAnchor.href.includes('github')) {
+      gitHost = 'github';
+    } else {
+      return dom;
+    }
+
+    const title = (contentContainer.querySelector(
+      'h1',
+    ) as HTMLElement).textContent!.replace(/[^a-zA-Z ]/g, '');
+    const issueTitle = encodeURIComponent(`Documentation Feedback: ${title}`);
+    const issueDesc = encodeURIComponent(
+      `Page source:\n${sourceAnchor.href}\n\nFeedback:`,
+    );
+    const sourceURL = new URL(sourceAnchor.href);
+    const repoPath = sourceURL.pathname.split('/').slice(0, 3).join('/');
+
+    const feedbackLink = sourceAnchor.cloneNode() as HTMLAnchorElement;
+    switch (gitHost) {
+      case 'gitlab':
+        feedbackLink.href = `${sourceURL.origin}${repoPath}/issues/new?issue[title]=${issueTitle}&issue[description]=${issueDesc}`;
+        break;
+      case 'github':
+        feedbackLink.href = `${sourceURL.origin}${repoPath}/issues/new?title=${issueTitle}&body=${issueDesc}`;
+        break;
+      default:
+        return dom;
+    }
+    feedbackLink.insertAdjacentHTML(
+      'afterbegin',
+      // 'Feedback' icon from https://material-ui.com/components/material-icons
+      '<svg <g style="width:22px"><path d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"></path></g></svg>',
+    );
```

Comment:
Oooh, a bit sad that we can't do this by importing the icon and not copying the svg. Maybe there is a way?

```
import FeedbackIcon from '@material-ui/icons/Feedback';
```

Reply:
Hmm perhaps material-ui/icons exposes the svg somewhere. I'll take a look

Reply:
Updated